OsmAnd-java/src/net/osmand/util/MapUtils.java

```diff
@@ -274,7 +281,29 @@ public int compare(MapObject o1, MapObject o2) {
 			}
 		});
 	}
-	
+
+    /* Share/SEND location to another app for a specific action (message, QRCode, etc) */
+	public static void shareLocation(Context ctx, double latitude, double longitude, int zoom) {
```

Comment:
This class could not use any Android classes. Also it is not really clear why do we avoid previous Dialog with QR code and direct Email/SMS selection